protocol.go

```diff
@@ -201,41 +189,46 @@ func (p *Protocol) clearWrites(start, end uint16) {
 	emptyBufferIndices(second)
 }
 
-func (p *Protocol) ReadPacket(header PacketHeader, buf []byte) []byte {
+func (p *Protocol) ReadPacket(header PacketHeader, buf []byte) (needed bool) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
 	p.readAckBits(header.ACK, header.ACKBits)
 
 	if !header.Unordered && !p.trackRead(header.Sequence) {
-		return nil
+		return
 	}
 
 	p.trackUnacked()
 
 	if header.Empty {
```

Comment:
warning: exported method Protocol.ReadPacket should have comment or be unexported (golint)